lib/core/src/text.rs

```diff
@@ -25,34 +37,85 @@ use web_sys::{WebGlRenderingContext,WebGlBuffer,WebGlTexture};
 /// commits
 #[derive(Debug)]
 pub struct TextComponent {
+    pub lines       : Vec<String>,
+    pub font        : FontId,
+    pub position    : Point2<f64>,
+    pub size        : Vector2<f64>,
+    pub text_size   : f64,
     gl_context      : WebGlRenderingContext,
     gl_program      : Program,
     gl_msdf_texture : WebGlTexture,
-    lines           : Vec<Line>,
     buffers         : TextComponentBuffers,
-    to_window       : Projective2<f64>
-}
-
-#[derive(Debug)]
-pub struct Line {
-    pub content                  : String,
 }
 
 impl TextComponent {
+    /// Scroll text by given offset.
+    ///
+    /// The value of 1.0 on both dimensions is equal to one line's height.
+    pub fn scroll(&mut self, offset:Vector2<f64>) {
+        self.buffers.scroll(offset);
+    }
+
+    /// Get current scroll position.
+    ///
+    /// The _scroll_position_ is a position of top-left corner of the first line.
+    /// The offset of 1.0 on both dimensions is equal to one line's height.
+    pub fn scroll_position(&self) -> &Vector2<f64> {
+        &self.buffers.scroll_offset
+    }
+
+    /// Jump to scroll position.
+    ///
+    /// The `scroll_position` is a position of top-left corner of the first line.
+    /// The offset of 1.0 on both dimensions is equal to one line's height.
+    pub fn jump_to_position(&mut self, scroll_position:Vector2<f64>) {
+        self.buffers.jump_to(scroll_position);
+    }
+
 
     /// Render text
-    pub fn display(&self) {
+    pub fn display(&mut self, fonts:&mut Fonts) {
         let gl_context     = &self.gl_context;
         let vertices_count = self.buffers.vertices_count() as i32;
+        let lines          = &mut self.lines;
+        let font           = fonts.get_render_info(self.font);
+        let content_ref    = ContentRef{lines,font};
 
+        self.buffers.refresh(gl_context,content_ref);
         gl_context.use_program(Some(&self.gl_program));
+        self.update_uniforms();
         self.bind_buffer_to_attribute("position",&self.buffers.vertex_position);
-        self.bind_buffer_to_attribute("texCoord",&self.buffers.texture_coords);
+        self.bind_buffer_to_attribute("tex_coord",&self.buffers.texture_coords);
         self.setup_blending();
         gl_context.bind_texture(Context::TEXTURE_2D, Some(&self.gl_msdf_texture));
         gl_context.draw_arrays(WebGlRenderingContext::TRIANGLES,0,vertices_count);
     }
 
+    fn update_uniforms(&self) {
+        let gl_context          = &self.gl_context;
+        let to_window           = self.create_to_window();
+        let to_window_ref       = to_window.as_ref();
+        let to_window_loc       = gl_context.get_uniform_location(&self.gl_program,"to_window");
+        let transpose           = false;
+        gl_context.uniform_matrix3fv_with_f32_array(to_window_loc.as_ref(),transpose,to_window_ref);
+    }
+
+    fn create_to_window(&self) -> SmallVec<[f32;9]> {
```

Comment:
What does `create_to_window` mean? Btw, please provide docs to all pub functions that are not getters and setters (we can agree that getters and setters dont need docs - can we add it to the guide?).